### PR TITLE
fix(es/loader): fix browser overwrites not applying correctly

### PR DESCRIPTION
**Pull request by icewind1991**
- es/loader: add test for browser overwrite
- es/loader: fix browser overwrites

**Description:**

Fix browser `main` overwrites from `package.json` not working when specified like

```json
{
  "name": "jquery",
  "main": "./index.js",
  "browser": {
    "./index.js": "./browser.js"
  }
}
```

Found in the wild in [solidjs](https://github.com/solidjs/solid/blob/main/packages/solid/web/package.json#L5)

This was mainly broken due to a mix of absolute and relative paths being compared.

### CODE REVIEW

File: crates/swc_ecma_loader/src/resolvers/node.rs [hunk 1]
```diff
@@ -295,12 +295,13 @@ impl NodeModulesResolver {
 
                                 // Relative file paths are sources for this package
                                 let source = if let Some(Component::CurDir) = components.next() {
+                                    dbg!(pkg_dir, k);
```
Comment (kdy1) [inline crates/swc_ecma_loader/src/resolvers/node.rs:298]: Please remove this line

File: crates/swc_ecma_loader/src/resolvers/node.rs [hunk 2]
```diff
@@ -482,13 +487,16 @@ impl Resolve for NodeModulesResolver {
             // Handle path references for the `browser` package config
             if let TargetEnv::Browser = self.target_env {
                 if let FileName::Real(path) = &v {
-                    if let Some(pkg_base) = find_package_root(base) {
+                    if let Some(pkg_base) = find_package_root(path) {
+                        let pkg_base = to_absolute_path(&pkg_base).unwrap();
                         if let Some(item) = BROWSER_CACHE.get(&pkg_base) {
                             let value = item.value();
-                            if value.ignores.contains(path) {
+                            dbg!(path);
```
Comment (kdy1) [inline crates/swc_ecma_loader/src/resolvers/node.rs:494]: Ditto

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thanks!

---

swc-bump: 
 - swc_ecma_loader

File: crates/swc_ecma_loader/src/resolvers/node.rs (hunk 1 above)
Comment (icewind1991) [reply crates/swc_ecma_loader/src/resolvers/node.rs]: oops :see_no_evil:

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_ecma_loader

**Pull request merged**